students/MikeShand/lesson04/mailroom_2.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+# donors = [['Andy', [10.00]], ['Bill', [15.00, 25.00]], ['Chuck', [20.00, 30.00, 40.00]]]
+
+donors = {'Andy': [10.00], 'Bill': [15.00, 25.00], 'Chuck': [20.00, 30.00, 40.00]}
+
+
+def donor_list():
+    for name in donors:
+        print(name)
+    thank_you()
+
+
+def thank_you():
+    print('Please enter the donor name\n (Type "list" for a list of current donor names)\n '
+          'Press "q" to return to console')
+    new_donor = input(':')
+
+    if new_donor.lower() == 'list':
+        donor_list()
+
+    elif new_donor.lower() == 'q':
+        return
+
+    else:
+
+        amount = float(input('Please enter the donation amount:'))
+
+        if new_donor in donors:
+            donors[new_donor].append(amount)
+
+        else:
+            donors[new_donor] = [amount]
+
+        print('Thank you {} for your generous donation of ${:.2f}'.format(new_donor, amount))
+
+
+def donor_report():
+    print('Here is a list of donors and contributions')
+    print('Name                 Total                Donations            Average')
+    for data in donors:

```

Comment:
this is technically for `name in donors`
since you also need the dict values, instead of accessing by key just loop over both keys and values (a more common pattern)
`for name, data in donors.items():`